### PR TITLE
ipfs: 0.8.0 → 0.9.0

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabianhjr** [comment]
Note:

```
2 packages failed to build:
python38Packages.ipfshttpclient python39Packages.ipfshttpclient
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127871` at d96ccfaf run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/787cdd3d87670b80a6c18ec5f094193a">python38Packages.ipfshttpclient</a></li>
    <li><a href="https://gist.github.com/853e46364d7de38bdfecf5d2016404f6">python39Packages.ipfshttpclient</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>ipfs</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by ElXreno** [comment]
Result of `nixpkgs-review pr 127871` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.ipfshttpclient</li>
    <li>python39Packages.ipfshttpclient</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ipfs</li>
  </ul>
</details>

File: pkgs/applications/networking/ipfs/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "ipfs";
-  version = "0.8.0";
+  version = "0.9.0";
   rev = "v${version}";
 
   # go-ipfs makes changes to it's source tarball that don't match the git source.
   src = fetchurl {
     url = "https://github.com/ipfs/go-ipfs/releases/download/${rev}/go-ipfs-source.tar.gz";
-    sha256 = "sha256-uK3+Ekr5AM6mmGmjFSj1Rotm5pbH657BYUlP9B39WEw=";
+    sha256 = "sha256:1fyffnw1d860w7gwm6ijbgrh68297z5bmvww8yqfshm3xgvcs6bf";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/ipfs/default.nix:11]:
```suggestion
    sha256 = "1fyffnw1d860w7gwm6ijbgrh68297z5bmvww8yqfshm3xgvcs6bf";
```

**Pull request merged**